### PR TITLE
getdns: add getdns/1.6.0 recipe

### LINKED ISSUE #2891: [request] getdns/1.6.0
### Package Details
  * Package Name/Version: **getdns/1.6.0**
  * Website: **https://getdnsapi.net/**
  * Source code: **https://github.com/getdnsapi/getdns**

### Description Of The Library / Tool

> getdns is an implementation of a modern asynchronous DNS API. It is intended to make all types of DNS information easily available to application developers and non-DNS experts.

I note that getdns switched to cmake in v1.6.0; might be easier to package now than previously.

> Mayor change from previous release is that the autotools based build system has been replaced with a CMake build system.
> 
> The motivation for this change is to have Windows be a first class build platform for getdns thereby increasing the ease of use of the getdns library in Windows applications. getdns and stubby can now be built using a native Windows build toolchain, ideally via the Visual Studio project.

- https://getdnsapi.net/releases/getdns-1-6-0/
- https://lektor.getdnsapi.net/quick-start/cmake-quick-start/

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **getdns/1.6.0**

Fixes #2891

- [x] Requires #2864 for libidn2
- [x] Requires #2902 for libevent pkg_config

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by uilianries — APPROVED** [review]

**Comment by madebr** [comment]
@uilianries @jgsogo @danimtb 
I cannot reproduce the symlink related error on my Windows box.
Maybe something special with the Windows CI servers?

**Review by danimtb — APPROVED** [review]

**Comment by jgsogo** [comment]
Hi! I can see this error

```
  CMake Error at source_subfolder/cmake_install.cmake:140 (file):
    file INSTALL cannot read symlink
    "C:/J/w/cci_PR-2903@3/.conan/data/getdns/1.6.0/_/_/build/4880f0ac5f1018186cb9fbc9fe296091ae8b22b1/source_subfolder/COPYING"
    to duplicate at
    "C:/J/w/cci_PR-2903@3/.conan/data/getdns/1.6.0/_/_/package/4880f0ac5f1018186cb9fbc9fe296091ae8b22b1/share/doc/getdns/COPYING":
    File exists.
  Call Stack (most recent call first):
    cmake_install.cmake:37 (include)
```

I'll try to reproduce it in our staging environment, there I can enter the Windows machine and run locally some commands, in production it is not possible.

**Comment by madebr** [comment]
@jgsogo 
So the cmake script contains the following line:
```cmake
install(FILES COPYING LICENSE)
```

But `COPYING` is a symlink to `LICENSE`.
```
lrwxrwxrwx.  1 maarten maarten     7 26 nov 23:51 COPYING -> LICENSE
-rw-rw-r--.  1 maarten maarten  1481 28 feb  2020 LICENSE
```
So it looks like the builders are stumbling on installing symbolic links on Windows.

**Comment by madebr** [comment]
@jgsogo @danimtb @uilianries 
I patched the cmake script to don't attempt to install the symlinked license.
So the problem is still there.

File: recipes/getdns/all/patches/1.6.0-0004-dont-install-symlinked-license.patch [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    "C:/J/w/cci_PR-2903@2/.conan/data/getdns/1.6.0/_/_/package/dd6d0e14f162a68e24ee52fee674d130eac57dea/share/doc/getdns/COPYING":
+    File exists.
+  Call Stack (most recent call first):
+
+
+--- CMakeLists.txt
++++ CMakeLists.txt
+@@ -1080,1 +1080,1 @@
+-install(FILES AUTHORS ChangeLog COPYING LICENSE NEWS README.md DESTINATION ${docdir})
++install(FILES AUTHORS ChangeLog NEWS README.md DESTINATION ${docdir})
```
Comment (joelpurra) [inline recipes/getdns/all/patches/1.6.0-0004-dont-install-symlinked-license.patch:16]: How about keeping the `LICENSE` file? Only the `COPYING` symlink is problematic, right?

(Not sure this makes any difference for the conan recipe; it is probably not the same file as in the license-packaging step, right?)

- https://github.com/conan-io/conan-center-index/pull/2903/files#diff-91193596606c700ab240a997decc0b2243d6eb6adca4ca3f020023a6d91e0c28R125

File: recipes/getdns/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        # Use FindOpenSSL.cmake to let check_function_exists succeed
+        # Remove other cmake modules as they use FindPkgConfig
+        for fn in glob.glob("Find*cmake"):
+            if "OpenSSL" not in fn:
+                os.unlink(fn)
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("COPYING", src=self._source_subfolder, dst="licenses")
```
Comment (joelpurra) [inline recipes/getdns/all/conanfile.py:125]: Should this be `LICENSE`, instead of the symlinked `COPYING`?

**Review by joelpurra — CHANGES_REQUESTED** [review]
Am not experienced with conan recipes, but should (explicitly) include the `LICENSE` _file_, right?

The `COPYING` _symlink_ seems to be targeted a the moment, although not sure if the packaging step copies the actual symlink or the target (and what the name of the resulting file is; that of the symlink or the target) when running on linux/macos/windows.

File: recipes/getdns/all/patches/1.6.0-0004-dont-install-symlinked-license.patch (hunk 1 above)
Comment (madebr) [reply recipes/getdns/all/patches/1.6.0-0004-dont-install-symlinked-license.patch:16]: I opted to do the safe way and remove both from the list.
The docdir is removed anyways.

File: recipes/getdns/all/conanfile.py (hunk 2 above)
Comment (madebr) [reply recipes/getdns/all/conanfile.py:125]: Indeed, but I didn't pass `symlinks=True` to `self.copy`.
So not the symlink is copied, but the destination file (=`LICENSE`).
So in the end it's ok.

**Review by joelpurra — APPROVED** [review]
Guess my suggestions make no practical difference, they're just for "correctness". (Only looked at the `COPYING` symlink though.)

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**